[CUBRIDMAN-298] [EN] [KO] 매뉴얼 구조 정리와 프로시저 등록에 대한 설명 강화 (#643)
CUBRID/cubrid-manual: en/pl/pl_create.rst

            today DATE DEFAULT SYS_DATE
        ) RETURN INT
        AS
        BEGIN
            RETURN YEAR(today) - YEAR(birth)
                - CASE WHEN TO_CHAR(today, 'MMDD') < TO_CHAR(birth, 'MMDD') THEN 1 ELSE 0 END;
        END;

        SELECT get_age();
        SELECT get_age(DATE'2000-05-10');
        SELECT get_age(DATE'2000-05-10', DATE'2025-03-24');

::        

        get_age()
        =============
                   25

        get_age(date '2000-05-10')
        ============================
                                  24

        get_age(date '2000-05-10', date '2025-03-24')
        ===============================================
                                                     24

.. _pl_function_overloading:

Function Overloading Not Supported
======================================

Function overloading is the process of defining multiple functions with the same name but different argument types or numbers.
**CUBRID does not support function overloading**. This means that you cannot create multiple stored procedures or stored functions with the same name if they have different argument types or numbers.

> Therefore, when using the **CREATE OR REPLACE** statement, the new procedure is not created and the existing procedure is overwritten, so caution is required.

Reviewer: ```suggestion
Therefore, when using the **CREATE OR REPLACE** statement, the new procedure will overwrite the existing procedure rather than create a new one; use it with caution to avoid accidental replacement.
```
이 문장에는 "the new procedure is not created"보다 "the existing procedure is overwritten"를 강조해야 덜 헷갈릴 듯 합니다. 또한 이 두 라인 사이에 "and"를 사용하는 것 보다 "rather than"을 사용하여 연결하는게 좋을 듯 합니다 (현재는 "프로시저가 새롭게 생성되지 않고 기존에 등록한 프로시저를 덮어쓴다" 라는 의미가 아닌 "새로운 프로시저가 아예 생성되지 않으며 기존에 등록한 프로시저도 덮어쓰인다/손상시킨다"라고 해석이 됩니다).

Therefore, if the table, view, sequence, or index objects referenced in the PL/CSQL procedure do not exist, an error will occur when registering the procedure.

.. code-block:: sql

    DROP TABLE IF EXISTS test_tbl;

    CREATE OR REPLACE FUNCTION test_func () RETURN INT
        k INT;
    AS
    BEGIN
        SELECT id INTO k FROM test_tbl LIMIT 1;
        RETURN k;
    END;

::

    ERROR: Stored procedure compile error: Syntax: before '  LIMIT 1'
    Unknown class "dba.test_tbl". select id from [dba.test_tbl] limit 1

The following example will create the test_tbl table correctly.

.. code-block:: sql

    DROP TABLE IF EXISTS test_tbl;
    CREATE TABLE test_tbl (id INT);
    INSERT INTO test_tbl VALUES (1);

    CREATE OR REPLACE FUNCTION test_func () RETURN INT
    AS
        k INT;
    BEGIN
        SELECT id INTO k FROM test_tbl LIMIT 1;
        RETURN k;
    END;

    SELECT test_func();

::

      test_func()
    =============
                1

In the case of PL/CSQL Dynamic SQL or Java SP, the stored string is executed at runtime, 

> so it is not checked when registering the procedure, but it is checked when calling the procedure.

Reviewer: ```suggestion
so it is not examined when registering the procedure, and is instead validated when the procedure is called.
```

.. code-block:: sql

    DROP TABLE IF EXISTS test_tbl;

    CREATE OR REPLACE FUNCTION test_func () RETURN INT
    AS
        k INT;
    BEGIN
        execute immediate 'SELECT id FROM test_tbl LIMIT 1' INTO k;
        RETURN k;
    END;

    SELECT test_func();

::

    ERROR: Stored procedure execute error: 
      (line 5, column 9) Syntax: before '  LIMIT 1'
    Unknown class "dba.test_tbl". select id from [dba.test_tbl] limit 1

> If the object referenced internally by the stored procedure or stored function is deleted, 

Reviewer: ```suggestion
If an object referenced internally by a stored procedure or stored function is deleted, 
```

> the system does not check the dependency, so even if it is registered correctly, an error may occur when the procedure is called.

Reviewer: ```suggestion
the system does not perform dependency checks, so even if the routine is registered correctly, an error may occur when the procedure is called.
```
